### PR TITLE
Load only the indicated command (if any), not everything in advance.

### PR DESCRIPTION
**Pull request by facundobatista**
For this I pre-parsed the sysargs, which moved ourselves to a MUCH SIMPLER use of ArgumentParser, as there are no more "commands" involved at that level.

Of course we're still using ArgumentParser, but only for each command.

Everything got MUCH MORE SIMPLE now, as there is quite less fiddling with ArgumentParser internals.

Sorry fo the length of the branch, but I improved several tests, added corner cases to others, and created a couple of new ones.

### CODE REVIEW

**Review by chipaca — APPROVED** [review]
noice

File: charmcraft/main.py [hunk 1]
```diff
@@ -181,91 +151,96 @@ class Dispatcher:
...
+                filtered_sysargs.append(arg)
 
-    def _handle_global_params(self):
-        """Set up and process global parameters."""
-        if _get_option(self.parsed_args, 'verbose'):
-            message_handler.set_mode(message_handler.VERBOSE)
-        if _get_option(self.parsed_args, 'quiet'):
+        # control and use quiet/verbose options
+        if flags['quiet'] and flags['verbose']:
+            print("Error, mutex!")
```
Comment (justinmclark) [inline charmcraft/main.py:211]: Should this print still be here?

**Review by justinmclark — APPROVED** [review]
+1

Just one comment/question
